content/blog/gatbsy-typescript/index.md

```diff
@@ -0,0 +1,284 @@
+---
+title: Adding Typescript to your Gatbsy application
+date: "2020-05-10"
+author: franleplant
+description: "Docker has become a widely used
+  technology and chances are you are going to have to deal with it eventually, at least superficially, in your Front End
+  career. Let's cover the basic concepts and day to day useful commands you will likely use when dealing with Docker."
+tags:
+  - gatsbyjs
+  - javascript
+  - react
+  - typescript
+  - graphql
+---
+
+## Intro
+
+I have been using Typescript for the past 4 years in different production
+projects in different companies and although this post is not meant to be
+a _why typescript?_ post let me tell you briefly why I love typescript
+and why I do think that most medium to largse size Javascript applications
+would benefit from using Typescript, in no particular order:
+
+1- automatically up-to-date and synchronized code documentation in the form of types.
+2- new code verification tool added to your pool (in javascript apps this is mostly made of different kinds of tests and lints)
+3- better development experience: better autocompletion and api discovery
+4- type checker guided hassle free refactoring
+5- improve the mind set of developers so that they think more about data structures and their semantic meaning inside the code (as oposed to just have a bunch of unamed and uncategorized objects which is something that tends to happen in bigger javascript apps).
+
+I plan to talk a lot more about this in a follow up post so that is why I wont focus to much
+explaining these points, but please note that most of these come from real world experiences
+handling large production code bases in both Javascript and Typescript, and there is an abysm between the too.
+
+Additionally, from a business standpoint we can also enumerate derived benefits:
+
+- more productivity: items 1, 3 and 4
+- higher quality: items 2, 4 and 5
+
+Having these in mind we are going to cover how to integrate Typescript to Gatsby,
+considering that Gatbsy is a framework that can accomodote larger production applications that
+will benefit the most out of this.
+
+The process is rather simple but I want to cover some paint points and tips and tricks.
+
+Note: This blog is actually using Typescript.
+
+## How to use Typescript in your Gatbsy.js site.
+
+We are going to install two plugins:
```

Comment:
first plugin appears to be included in Gatsby which might be worth mentioning

Reply:
it is included in the `gatsby-starter-blog`, Im trying to make this more general to any gatsby application